### PR TITLE
Basic Google Analytics tracking for URL changes in workbench.

### PR DESCRIPTION
**Pull request by danqrodney**
Adding emitEvent() function for future use.

### CODE REVIEW

File: ui/src/app/google-analytics-events.service.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import {Injectable} from "@angular/core";
+declare let gtag: Function;
+ 
+@Injectable()
+export class GoogleAnalyticsEventsService {
+ 
+  public emitEvent(eventName: string,
+		   eventCategory: string,
```
Comment (johnstill) [inline ui/src/app/google-analytics-events.service.ts]: NIT: indent one more space plz

File: ui/src/app/google-analytics-events.service.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+                   eventAction: string,
+                   eventLabel: string = null,
+                   eventValue: number = null) {
+    gtag('event', eventName, {
+      eventCategory: eventCategory,
```
Comment (johnstill) [inline ui/src/app/google-analytics-events.service.ts]: When writing an object literal, if the key and value share the same name they can be collapsed: ie

```
const eventLabel = 'some label';
const obj = {eventLabel};  // equivalent to obj = {eventLabel: eventLabel};
```

So this becomes shortened to

```
{ eventCategory, eventLabel, eventAction, eventValue }
```

File: ui/src/app/google-analytics-events.service.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      eventLabel: eventLabel,
+      eventAction: eventAction,
+      eventValue: eventValue
+    });
+  }
+}
```
Comment (johnstill) [inline ui/src/app/google-analytics-events.service.ts]: NIT: end the files with a newline please.

File: ui/src/index.html [hunk 4]
```diff
@@ -21,6 +21,17 @@
         google.charts.load('current', {'packages': ['corechart']});
     </script>
 
+    <!-- Global site tag (gtag.js) - Google Analytics -->
+    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-112406425-1"></script>
+   <script>
```
Comment (johnstill) [inline ui/src/index.html]: NIT: this tag and the closing `<script>` tag need to be indented another space plz.

File: ui/src/index.html [hunk 5]
```diff
@@ -21,6 +21,17 @@
...
+       window.dataLayer = window.dataLayer || [];
```
Comment (blrubenstein) [inline ui/src/index.html]: Nit: Spacing. (This is indented in four spaces from the script tag, I believe we are doing two? I realized we didn't do that for the charts, I'll go back and fix that as a follow up)

File: ui/src/app/google-analytics-events.service.ts (hunk 1 above)
Comment (danqrodney) [reply ui/src/app/google-analytics-events.service.ts]: Done.

File: ui/src/app/google-analytics-events.service.ts (hunk 2 above)
Comment (danqrodney) [reply ui/src/app/google-analytics-events.service.ts]: Done.

File: ui/src/app/google-analytics-events.service.ts (hunk 3 above)
Comment (danqrodney) [reply ui/src/app/google-analytics-events.service.ts]: Done.

File: ui/src/index.html (hunk 4 above)
Comment (danqrodney) [reply ui/src/index.html]: Done.

File: ui/src/index.html (hunk 5 above)
Comment (danqrodney) [reply ui/src/index.html]: Changed to two in both places.

**Pull request merged**